go-controller/pkg/cni/cni.go

```diff
@@ -116,8 +116,59 @@ func (pr *PodRequest) cmdDel() ([]byte, error) {
 	return []byte{}, nil
 }
 
-func (pr *PodRequest) cmdCheck() ([]byte, error) {
-	// TODO add implementation
+func (pr *PodRequest) cmdCheck(podLister corev1listers.PodLister) ([]byte, error) {
+	namespace := pr.PodNamespace
+	podName := pr.PodName
+	if namespace == "" || podName == "" {
+		return nil, fmt.Errorf("required CNI variable missing")
+	}
+	ifaceID := fmt.Sprintf("%s_%s", namespace, podName)
+
+	// Get the IP address and MAC address of the pod
+	annotations, err := getPodAnnotations(pr.ctx, podLister, pr.PodNamespace, pr.PodName)
+	if err != nil {
+		return nil, err
+	}
+
+	ingress, egress, err := extractPodBandwidthResources(annotations)
+	if err != nil {
+		return nil, fmt.Errorf("failed to parse bandwidth request: %v", err)
+	}
+
+	if pr.CNIConf.PrevResult != nil {
+		result, err := current.NewResultFromResult(pr.CNIConf.PrevResult)
+
+		if err != nil {
+			return nil, fmt.Errorf("could not convert result to current version: %v", err)
+		}
+		hostInterfaceIdx := -1
```

Comment:
Nit: for the sake of clarity when reading the code: I would change this to a `bool`. I thought the `hostInterfaceIdx` was being used later on when going over this commit. But it's only assessing if the `hostIfaceName` has been found or not

Reply:
on the second thought, I can remove it at all, and just check that `hostIfaceName` is not empty.